install/setup-js-sdk-assets.sh

```diff
@@ -21,16 +21,29 @@ if [[ "${SETUP_JS_SDK_ASSETS:-}" == "1" ]]; then
   # We want to remove everything before the first '{'.
   loader_registry=$(echo "$loader_registry" | sed '0,/{/s/[^{]*//')
 
+  #  Sentry backend provides SDK versions from v4.x up to v8.x.
+  latest_js_v4=$(echo "$loader_registry" | $jq -r '.versions | reverse | map(select(.|any(.; startswith("4.")))) | .[0]')
+  latest_js_v5=$(echo "$loader_registry" | $jq -r '.versions | reverse | map(select(.|any(.; startswith("5.")))) | .[0]')
+  latest_js_v6=$(echo "$loader_registry" | $jq -r '.versions | reverse | map(select(.|any(.; startswith("6.")))) | .[0]')
   latest_js_v7=$(echo "$loader_registry" | $jq -r '.versions | reverse | map(select(.|any(.; startswith("7.")))) | .[0]')
   latest_js_v8=$(echo "$loader_registry" | $jq -r '.versions | reverse | map(select(.|any(.; startswith("8.")))) | .[0]')
 
-  echo "Found JS SDKs v${latest_js_v7} and v${latest_js_v8}, downloading from upstream.."
+  echo "Found JS SDKs: v${latest_js_v4}, v${latest_js_v5}, v${latest_js_v6}, v${latest_js_v7}, v${latest_js_v8}"
 
   # Download those two using wget
-  for version in "${latest_js_v7}" "${latest_js_v8}"; do
+  for version in "$latest_js_v4" "$latest_js_v5" "$latest_js_v6" "$latest_js_v7" "$latest_js_v8"; do
     $dcr --no-deps --rm -v "sentry-nginx-www:/var/www" nginx mkdir -p /var/www/js-sdk/${version}
-    for variant in "tracing" "tracing.replay" "replay" "tracing.replay.feedback" "feedback"; do
-      $dcr --no-deps --rm -v "sentry-nginx-www:/var/www" nginx wget -q -O /var/www/js-sdk/${version}/bundle.${variant}.min.js "https://browser.sentry-cdn.com/${version}/bundle.${variant}.min.js"
+    for variant in "bundle" "bundle.tracing" "bundle.tracing.replay" "bundle.replay" "bundle.tracing.replay.feedback" "bundle.feedback"; do
+      # We want to have a HEAD lookup. If the response status code is not 200, we will skip the variant.
+      # Taken from https://superuser.com/questions/272265/getting-curl-to-output-http-status-code#comment1025992_272273
+      status_code=$($dcr --no-deps --rm nginx curl --retry 3 --silent -I "https://browser.sentry-cdn.com/${version}/${variant}.min.js" 2>/dev/null | head -n 1 | cut -d$' ' -f2)
+      if [[ "$status_code" != "200" ]]; then
+        echo "Skipping download of JS SDK v${version} for ${variant}.min.js, because the status code was ${status_code} (non 200)"
+        continue
+      fi
```

Comment:
Can we not use the `-f/--fail` flag yet (https://superuser.com/a/657174)

I think it was added in a newer version of `curl` so not sure if it would be available here. Would be great if it is though.

Reply:
The nginx image has the curl version of 8.1.x. Should be fine.